Fix gcc install errors

I just did a fresh install of the GMT and got these warnings.

```
gcc ../../../../../lib/c/parse_int.o source.c -O3 -Wall -I../../../../../lib/c -o metric-provider-binary
source.c: In function ‘get_memory_procfs’:
source.c:35:5: warning: ignoring return value of ‘fgets’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
   35 |     fgets(buf, 200, fd);
      |     ^~~~~~~~~~~~~~~~~~~
source.c:42:5: warning: ignoring return value of ‘fgets’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
   42 |     fgets(buf, 200, fd); // drop MemFree
      |     ^~~~~~~~~~~~~~~~~~~
source.c:44:5: warning: ignoring return value of ‘fgets’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
   44 |     fgets(buf, 200, fd);
      |     ^~~~~~~~~~~~~~~~~~~
make: Leaving directory '/home/didi/green-metrics-tool/metric_providers/memory/used/procfs/system'
Installing ./metric_providers/network/io/cgroup/container/metric-provider-binary ...
make: Entering directory '/home/didi/green-metrics-tool/metric_providers/network/io/cgroup/container'
gcc ../../../../../lib/c/parse_int.o source.c -O3 -Wall -lc -I../../../../../lib/c -o metric-provider-binary
source.c: In function ‘get_network_cgroup’:
source.c:85:5: warning: ignoring return value of ‘fgets’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
   85 |     fgets(buf, 200, fd);
      |     ^~~~~~~~~~~~~~~~~~~
source.c:86:5: warning: ignoring return value of ‘fgets’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
   86 |     fgets(buf, 200, fd);
      |     ^~~~~~~~~~~~~~~~~~~
sudo chown root metric-provider-binary
sudo chmod u+s metric-provider-binary
make: Leaving directory '/home/didi/green-metrics-tool/metric_providers/network/io/cgroup/container'
Installing ./metric_providers/network/io/procfs/system/metric-provider-binary ...
make: Entering directory '/home/didi/green-metrics-tool/metric_providers/network/io/procfs/system'
gcc ../../../../../lib/c/parse_int.o source.c -O3 -Wall -lc -I../../../../../lib/c -o metric-provider-binary
source.c: In function ‘output_network_procfs’:
source.c:52:5: warning: ignoring return value of ‘fgets’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
   52 |     fgets(buf, 200, fd);
      |     ^~~~~~~~~~~~~~~~~~~
source.c:53:5: warning: ignoring return value of ‘fgets’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
   53 |     fgets(buf, 200, fd);
      |     ^~~~~~~~~~~~~~~~~~~
make: Leaving directory '/home/didi/green-metrics-tool/metric_providers/network/io/procfs/system'

Building sgx binaries
make: Entering directory '/home/didi/green-metrics-tool/lib/sgx-software-enable'
gcc -c sgx_enable.c 
gcc -o sgx_enable sgx_enable.o
make: Leaving directory '/home/didi/green-metrics-tool/lib/sgx-software-enable'
```

--- a/metric_providers/memory/used/procfs/system/source.c
+++ b/metric_providers/memory/used/procfs/system/source.c
@@ -31,17 +31,21 @@ static unsigned long long int get_memory_procfs() {
         exit(1);
     }
 
+    if (fgets(buf, 200, fd) == NULL) {
+        fprintf(stderr, "Error or EOF encountered while reading input.\n");
+        exit(1);
+    }
 
-    fgets(buf, 200, fd);
     match_result = sscanf(buf, "MemTotal: %llu kB", &mem_total);
     if (match_result != 1) {
         fprintf(stderr, "Error - MemTotal could not be matched in /proc/meminfo\n");
         exit(1);
     }
 
-    fgets(buf, 200, fd); // drop MemFree
-
-    fgets(buf, 200, fd);
+    if (fgets(buf, 200, fd) == NULL || fgets(buf, 200, fd) == NULL) {
+        fprintf(stderr, "Error or EOF encountered while reading input.\n");
+        exit(1);
+    }
     match_result = sscanf(buf, "MemAvailable: %llu kB", &mem_available);
     if (match_result != 1) {
         fprintf(stderr, "Error - MemAvailable could not be matched in /proc/meminfo\n");

--- a/metric_providers/network/io/cgroup/container/source.c
+++ b/metric_providers/network/io/cgroup/container/source.c
@@ -81,9 +81,10 @@ static net_io_t get_network_cgroup(unsigned int pid) {
         exit(1);
     }
 
-    // skip first two lines
-    fgets(buf, 200, fd);
-    fgets(buf, 200, fd);
+    if (fgets(buf, 200, fd) == NULL || fgets(buf, 200, fd) == NULL) {
+        fprintf(stderr, "Error or EOF encountered while reading input.\n");
+        exit(1);
+    }
 
     int match_result = 0;
 
@@ -190,7 +191,7 @@ static int check_system(int rootless_mode) {
     } else {
         file_path_cgroup_procs = "/sys/fs/cgroup/system.slice/cgroup.procs";
     }
-    
+
     FILE* fd = fopen(file_path_cgroup_procs, "r");
     if (fd == NULL) {
         fprintf(stderr, "Couldn't open cgroup.procs file at %s\n", file_path_cgroup_procs);
@@ -269,7 +270,7 @@ int main(int argc, char **argv) {
     }
 
     if(check_system_flag){
-        exit(check_system(rootless_mode)); 
+        exit(check_system(rootless_mode));
     }
 
     int length = parse_containers(&containers, containers_string, rootless_mode);

--- a/metric_providers/network/io/procfs/system/source.c
+++ b/metric_providers/network/io/procfs/system/source.c
@@ -49,8 +49,10 @@ static void output_network_procfs() {
     }
 
     // skip first two lines
-    fgets(buf, 200, fd);
-    fgets(buf, 200, fd);
+    if (fgets(buf, 200, fd) == NULL || fgets(buf, 200, fd) == NULL) {
+        fprintf(stderr, "Error or EOF encountered while reading input.\n");
+        exit(1);
+    }
 
     int match_result = 0;
 

